Formula/earthly.rb

```diff
@@ -0,0 +1,32 @@
+class Earthly < Formula
+  desc "Build automation tool for the container era"
+  homepage "https://github.com/vladaionescu/earthly"
+  url "https://github.com/vladaionescu/earthly/archive/v0.1.1.tar.gz"
+  sha256 "17ce130e957739da1b96245235aebb584a54241f69aea0802b97948172f5744e"
+  head "https://github.com/vladaionescu/earthly.git"
+
+  bottle do
+    cellar :any_skip_relocation
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = HOMEBREW_CACHE/"go_cache"
+    system "go", "build", "-o", bin/"earth", "-ldflags",
+        "-X main.DefaultBuildkitdImage=earthly/buildkitd:v0.1.1 -X main.Version=v0.1.1 -linkmode external",
+        "./cmd/earth/main.go"
+  end
+
+  def caveats
+    <<~EOS
+      Visit GitHub repository for documentation:
+        https://github.com/vladaionescu/earthly
+    EOS
+  end
+
+  test do
+    output = shell_output("#{bin}/earth 2>&1", 1).split("\n")[1].strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.